### PR TITLE
Update classification-introduction

### PR DESCRIPTION
**Pull request by YinYi000**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] [local build the book](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-book)
- [ ] review the artifacts from local build
- [ ] [local build the slides](https://github.com/open-academy/machine-learning/blob/main/CONTRIBUTING.md#build-the-slides-optional) if appliable
- [ ] Unit test if appliable

# Checklist:

- [x] My code follows the [style guidelines](https://github.com/open-academy/machine-learning/blob/main/STYLE_GUIDE.md) of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have applied tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: open-machine-learning-jupyter-book/assignments/ml-fundamentals/classification_introduction.ipynb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+   "version": 3
+  },
+  "orig_nbformat": 2
+ },
+ "nbformat": 4,
+ "nbformat_minor": 2,
+ "cells": [
+  {
+   "source": [
+    "# Delicious Asian and Indian Cuisines "
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/ml-fundamentals/classification_introduction.ipynb:22]: Only first letter supposes to be in uppercase. And let's make the file name to be consistent with the content title `delicious-asian-and-indian-cuisines`

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 2]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+Image by [Jen Looper](https://twitter.com/jenlooper)
+```
+
+Classification is a form of [supervised learning](https://wikipedia.org/wiki/Supervised_learning) that bears a lot in common with regression techniques. If machine learning is all about predicting values or names to things by using datasets, then classification generally falls into two groups: _binary classification_ and _multiclass classification_.
+
+```{seealso}
+<div class="yt-container">
+  <iframe src="https://www.youtube.com/embed/eg8DJYwdMyg" allowfullscreen></iframe>
+</div>
+Click the video above for a quick introduction to classification.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:34]: Add an empty line between the text and div

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 3]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+
+```{note}
+Take a moment to imagine a dataset about cuisines. What would a multiclass model be able to answer? What would a binary model be able to answer? What if you wanted to determine whether a given cuisine was likely to use fenugreek? What if you wanted to see if, given a present of a grocery bag full of star anise, artichokes, cauliflower, and horseradish, you could create a typical Indian dish?
+```
+
+```{seealso}
+<div class="yt-container">
+  <iframe src="https://www.youtube.com/embed/GuTeDbaNoEU" allowfullscreen></iframe>
+</div>
+Click the video above. The whole premise of the show 'Chopped' is the 'mystery basket' where chefs have to make some dish out of a random choice of ingredients. Surely a ML model would have helped!
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:70]: Add an empty line between the text and div

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 4]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+
+## Hello 'classifier'
+
+The question we want to ask of this cuisine dataset is actually a **multiclass question**, as we have several potential national cuisines to work with. Given a batch of ingredients, which of these many classes will the data fit?
+
+Scikit-learn offers several different algorithms to use to classify data, depending on the kind of problem you want to solve. In the next two sections, you'll learn about several of these algorithms.
+
+## Exercise - clean and balance your data
+
+The first task at hand, before starting this project, is to clean and **balance** your data to get better results. Start with the blank [classification_introduction.ipynb](../../assignments/ml-fundamentals/classification_introduction.ipynb) file.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:81]: Please fix the ipynb link

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 5]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+
+The first thing to install is [imblearn](https://imbalanced-learn.org/stable/). This is a Scikit-learn package that will allow you to better balance the data (you will learn more about this task in a minute).
+
+1 . Import the packages you need to import your data and visualize it, also import `SMOTE` from `imblearn`.
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:85]: you can do `1\.`

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 6]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+import pandas as pd
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+import numpy as np
+from imblearn.over_sampling import SMOTE
+```
+
+Now you are set up to read import the data next.
+
+2 . The next task will be to import the data:
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:97]: similar to above

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 7]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+```
+
+7 . Now, drop the most common ingredients that create confusion between distinct cuisines, by calling `drop()`:
+
+Everyone loves rice, garlic and ginger!
+
+```{code-cell}
+:tags: [output_scroll]
+
+feature_df= df.drop(['cuisine','Unnamed: 0','rice','garlic','ginger'], axis=1)
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:208]: add spaces before the `,`

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 8]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+```{code-cell}
+transformed_df = pd.concat([transformed_label_df,transformed_feature_df],axis=1, join='outer')
+```
+
+4 . You can take one more look at the data using `transformed_df.head()` and `transformed_df.info()`. Save a copy of this data for use in future sections:
+
+```{code-cell}
+transformed_df.head()
+transformed_df.info()
+transformed_df.to_csv("../../assets/data/cleaned_cuisines.csv")
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:246]: You don't need to have this file in git. Please add it to the .gitignore.

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md [hunk 9]
```diff
@@ -14,3 +14,250 @@ kernelspec:
...
+```
+
+---
+
+## Self study
+
+This curriculum contains several interesting datasets. Dig through the `data` folders and see if any contain datasets that would be appropriate for binary or multi-class classification? What questions would you ask of this dataset?
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:253]: `data` should be our github repo's data folder name

**Review by zhangqi444 — COMMENTED** [review]
Please make sure this file could be built successfully from your local

**Review by zhangqi444 — APPROVED** [review]

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md (hunk 8 above)
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:250]: I don't understand, how to do it? what is .gitignore?

File: open-machine-learning-jupyter-book/assignments/ml-fundamentals/delicious-asian-and-indian-cuisines.ipynb [hunk 10]
```diff
@@ -0,0 +1,39 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Delicious asian and indian cuisines "
```
Comment (zhangqi444) [inline open-machine-learning-jupyter-book/assignments/ml-fundamentals/delicious-asian-and-indian-cuisines.ipynb:8]: '# Delicious Asian and Indian cuisines'

**Review by zhangqi444 — CHANGES_REQUESTED** [review]

**Review by zhangqi444 — CHANGES_REQUESTED** [review]

File: open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md (hunk 8 above)
Comment (YinYi000) [reply open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md:250]: which file should be added to the .gitignore.?  what is the .gitignore. use for ?

Comment (zhangqi444) [reply open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md]: ../../assets/data/cleaned_cuisines.csv -> ./cleaned_cuisines.csv

Then add `open-machine-learning-jupyter-book/ml-fundamentals/classification/cleaned_cuisines.csv` to gitignore

Comment (zhangqi444) [reply open-machine-learning-jupyter-book/ml-fundamentals/classification/introduction-to-classification.md]: The goal is to exclude this dynamically generated file from the git diff.

**Review by zhangqi444 — APPROVED** [review]

**Pull request merged**